internal/repository/rsl.go

```diff
@@ -33,6 +33,20 @@ func (r *Repository) RecordRSLEntryForReference(refName string, signCommit bool)
 		return err
 	}
 
+	// Check if the latest unskipped entry for the ref has the same target ID
+	// Note that it's legal for the RSL to have target A, then B, then A again,
+	// this is not considered a duplicate entry
+	latestUnskippedEntry, _, err := rsl.GetLatestUnskippedReferenceEntryForRef(r.r, absRefName)
+	if err == nil {
+		if latestUnskippedEntry.TargetID == ref.Hash() {
+			// duplicate entry, do nothing
+			return nil
+		}
+	} else if !errors.Is(err, rsl.ErrRSLEntryNotFound) {
+		// a prior entry may not exist, so that's the only allowed error
+		return err
+	}
+
```

Comment:
Given that this code is added twice (even with the same comment), would it make sense to refactor it into a `Repository` private helper function instead? Something like `isLatestEntry(refName string) bool`

Reply:
Done!